### ISSUE TITLE
Results of chain not passed to chord's header task

### ISSUE THREAD

**Issue by p-col** [celery/celery#3902]
## Checklist

- [X] I have included the output of ``celery -A proj report`` in the issue.
   (if you are not able to do this, then at least specify the Celery
   version affected).
- [X] I have verified that the issue exists against the `master` branch of Celery.

```
software -> celery:4.0.2 (latentcall) kombu:4.0.2 py:2.7.13
            billiard:3.5.0.2 py-amqp:2.1.4
platform -> system:Linux arch:64bit imp:CPython
loader   -> celery.loaders.app.AppLoader
settings -> transport:amqp results:db+postgresql://user:**@localhost/results

CELERY_TASK_SERIALIZER: 'json'
CELERY_RESULT_SERIALIZER: 'json'
CELERY_ACCEPT_CONTENT: ['application/json']
CELERY_INCLUDE: ['tasks']
CELERY_TIMEZONE: 'Europe/Paris'
BROKER_URL: u'amqp://myuser:********@127.0.0.1:5672/myvhost'
CELERY_RESULT_BACKEND: u'db+postgresql://user:********@localhost/results'
CELERYD_POOL_RESTARTS: True

```

## Steps to reproduce

Declare basic tasks like:

```
@celery.task
def add(a, b):
  return a+b

@celery.task
def tsum(nbs):
  return sum(nbs)
```

then build the signature and run it:

```
sig = add.s(0,1) | group([add.s(2), add.s(3) | group([add.s(4), add.s(5)])]) | tsum.s()
async_res = sig.apply_async()
print async_res.get()
```

## Expected behavior

This code is supposed to do:

- 0+1 = 1 then forward 1 to add.s(2) and add.s(3)
- 2+1 = 3 and 3+1 = 4 then forward 4 to add.s(4) and add.s(5)
- 4+4 = 8 and 4+5 = 9 etc.

## Actual behavior

This code hang forever probably because of result backend reasons but more importantly I got the following error from flower:

> Traceback (most recent call last):
>   File "/usr/local/lib/python2.7/dist-packages/celery-4.0.2-py2.7.egg/celery/app/trace.py", line 367, in trace_task
>     R = retval = fun(*args, **kwargs)
>   File "/usr/local/lib/python2.7/dist-packages/celery-4.0.2-py2.7.egg/celery/app/trace.py", line 622, in __protected_call__
>     return self.run(*args, **kwargs)
>
> TypeError: add() takes exactly 2 arguments (1 given)

And this is true because the received args are:
args    (3,)
instead of 1 from add.s(0,1) and 3 from the signature declaration add.s(3)

### TLDR;
Results are not passed correctly in chain of chord containing a chord
task | group(task, **task** | group | task) | task

**Comment by auvipy** [comment]
@p-col could you check #4481 and #3968
